lib/cartopy/mpl/gridliner.py

```diff
@@ -447,10 +448,13 @@ def __init__(self, axes, crs, draw_labels=False, xlocator=None,
         if auto_update is None:
             auto_update = True
         else:
+            # Note #2394 should be addressed before this deprecation expires.
             warnings.warn(
                 "The auto_update parameter was deprecated at Cartopy 0.23.  In future "
                 "the gridlines and labels will always be updated.",
-                DeprecationWarning)
+                DeprecationWarning,
+                stacklevel=(3 if sys._getframemodulename(1) == 'cartopy.mpl.geoaxes'
```

Comment:
I'm not sure we should be using the private method here. The documentation says it may not be available in all reference Python implementations.
https://docs.python.org/3/library/sys.html#sys._getframemodulename

I think you could possibly use the inspect module instead, but it looks like in other code locations we've just explicitly set the stacklevel to 2, is that enough to handle this and not special case different stack levels?

Reply:
Thanks - I didn't know `inspect` had something for that.  Now updated.  If we leave the stacklevel at 2 then the warning won't show for users who are calling `ax.gridlines`.  I was semi-following Matplotlib's `warn_external` which [uses sys._getframe](https://github.com/matplotlib/matplotlib/blob/b02aa82f3d3ba7f55e8a3fd431bc8763e3ae5c5c/lib/matplotlib/_api/__init__.py#L369).